### PR TITLE
Closes #832: Documentation for Full-Text Indexing(ElasticSearch)

### LINKED ISSUE #832: Documentation for Full-Text Indexing(ElasticSearch)
**What would you like to be added**:
PR for #537 is going to go in, we'll need documentation on how to use it.

**Why would you like this to be added**:
In case any future backend developments that depend on ES, contributors should have information on how to use the client as well as important params for the API.

### PR DESCRIPTION
**Pull request by manekenpix**
## Issue This PR Addresses

Closes #832 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [x] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Documentation for environment setup and usage.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [x] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: docs/elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- For listing all the indexes (`Browser`):
+
+```sh
+http://localhost:9200/_cat/indices?v
+```
+
+- You can query Elasticsearch using URL parameters (`Browser`):
+
+```sh
+http://localhost:9200/posts/_search?q=text:open%20source
```
Comment (c3ho) [inline docs/elasticsearch.md]: Should we also link the search api params [here ](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-search-api-query-params)and teach them how to use or combine the different parameters in the url/API
such as `http://localhost:9200/posts/_search?q=text:open%20source&size=10&from=1` to search the text `open source` and return only 10 indexes after skipping the first result?

File: docs/environment-setup.md [hunk 2]
```diff
@@ -6,6 +6,9 @@
 - [Redis](https://redis.io/) (2 methods)
   - Use [Docker and docker-compose](https://docs.docker.com/install/)
   - Install as a [native application](https://redis.io/topics/quickstart)
+- [Elasticsearch](https://www.elastic.co/what-is/elasticsearch) (2 methods)
+  - Use [Docker and docker-compose](https://docs.docker.com/install/)
+  - Install as a [native application](https://www.elastic.co/guide/en/elastic-stack-get-started/7.6/get-started-elastic-stack.html#install-elasticsearch)
```
Comment (c3ho) [inline docs/environment-setup.md]: I'd also put somewhere in here that we also need ES to be running in order for telescope to not crash, I'm not sure if we mention that for Redis as well. @smilegodly was asking why he was getting a protocol error which was probably caused by this.

File: docs/elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+[Jest Manual Mocks](https://jestjs.io/docs/en/manual-mocks) to provide mocks
+for Elasticsearch.
+
+For information about installing Redis, see [our environment setup guide](environment-setup.md).
+
+## Elasticsearch Tips
+
+- Interact with Elasticsearch outside of Telescope using one of the [available clients](https://www.elastic.co/guide/en/elasticsearch/client/index.html).
+- You can also interact with Elasticsearch's exposed `REST API` using a browser or [cURL](https://curl.haxx.se/).
+- If you want to see the state of the cluster (`cURL`):
```
[the remark this answers is not in the document]
Comment (manekenpix) [reply docs/elasticsearch.md]: It's possible to run cURL on windows, you just need to install it. The [cURL link](https://curl.haxx.se/) I added in line 14 has a `download` section with windows versions available.

File: docs/environment-setup.md (hunk 2 above)
Comment (manekenpix) [reply docs/environment-setup.md]: Added a warning in line 13.

File: docs/elasticsearch.md (hunk 1 above)
Comment (manekenpix) [reply docs/elasticsearch.md]: I added a link to the list of parameters. I think the 2 examples (line 30 and line 36) should be enough to know how to use the parameters.

**Comment by c3ho** [comment]
Ty for changes!

File: docs/elasticsearch.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+```sh
+ curl -X GET 'localhost:9200/\_cluster/health?pretty'
```
Comment (cindyledev) [inline docs/elasticsearch.md]: typo here, should be `/_cluster/` not `/\_cluster/`

Comment (manekenpix) [reply docs/elasticsearch.md]: I saw you approved the PR. Should I change this or it's ok?

Comment (manekenpix) [reply docs/elasticsearch.md]: Fixed

**Pull request merged**